modifications/DataClass.cc

```diff
@@ -61,21 +72,21 @@ void DataClass::readyReadToken()
 {
     QNetworkReply* reply = qobject_cast<QNetworkReply*>(sender());
     QString replyStr = reply->readAll();
-    QJsonDocument jsonResponse = QJsonDocument::fromJson(replyStr.toUtf8());
-    QJsonObject jsonObject = jsonResponse.object();
-    accessToken = jsonObject["access_token"].toString();
-    if(accessToken!=""){
-        emit tokenGenerated();
-    }
-    else{
+    int statusCode = reply->attribute( QNetworkRequest::HttpStatusCodeAttribute).toInt();
+    if(statusCode == 201 || statusCode == 200){
+        QJsonDocument jsonResponse = QJsonDocument::fromJson(replyStr.toUtf8());
+        QJsonObject jsonObject = jsonResponse.object();
+        accessToken = jsonObject["access_token"].toString();
+        if(accessToken!=""){
+            emit tokenGenerated();
+        }
+    }else{
         emit tokenNotGenerated();
+        showMessage("Unable to Connect to Server", statusCode);
```

Comment:
Minor but connect and server need not be in capital, also if 201 and 200 checks are to be performed elsewhere, might be nice to have them as a #define/enum on top. Would help avoid having magic numbers in the code even though these are well known status codes.

Reply:
Done, error codes are already an enum, no need to define in a separate header file because all API calls are made using dataClass